tests/tests_pytorch/callbacks/test_lambda_function.py

```diff
@@ -49,7 +49,7 @@ def call(hook, *_, **__):
         callbacks=[LambdaCallback(**hooks_args)],
     )
     with pytest.deprecated_call(
-        match="`on_init_start` callback hook was deprecated in v1.6 and will be removed in v1.8."
+        match="`on_configure_sharded_model` callback hook was deprecated in v1.6 and will be removed in v1.8"
```

Comment:
@AndresAlgaba Here is the fix for the failing test.

There are two deprecated methods overridden here. Since we are removing on_init_start, now the other one is still surfacing and we have to catch it. It will be removed by this PR eventually: https://github.com/Lightning-AI/lightning/pull/14834